api/envoy/config/filter/http/router/v2/router.proto

```diff
@@ -26,4 +26,7 @@ message Router {
   // an upstream request. Presuming retries are configured, multiple upstream
   // requests may be made for each downstream (inbound) request.
   repeated envoy.config.filter.accesslog.v2.AccessLog upstream_log = 3;
+
+  // Do not add any additional *x-envoy-* headers to requests or responses.
+  bool suppress_envoy_headers = 4;
```

Comment:
Does this means the suppression happens in router filter, so the other filter in the middle between HCM and router will not see response header *x-envoy-*?

Also from below impls seems router is not suppressing *x-envoy-* header that are generated in filters (e.g. grpc_json_transcoder generates *x-envoy-original-path*).

Reply:
It only means that the router filter itself won't generate any `x-envoy-` headers. Other filters still need to be configured to block as needed. The idea is that we can, in general, add other filters into the pipeline to filter `x-envoy-*` as needed, but not at the router filter, since it is the last in the filter chain, so we need something like this as at a minimum.

Reply:
Can you comment this explicitly only affects the router filter and that other parts of Envoy code will continue to set x-envoy headers?   I mistook suppress for remove as you called out on my test suggestion.